lib/grnds/sso/jwt/uid.rb

```
@@ -0,0 +1,31 @@
+module Grnds
+  module Sso
+    module Jwt
+      HEADER = 'Authorization'.freeze
+      COOKIE = 'user_auth0_token'.freeze
+      UID_CLAIM = :'https://grandrounds.com/email'.freeze
+
+      class Uid
+        class << self
+          def call(request)
+            request_token = token_from(request)
+            return unless request_token.present?
```

We expect returning nil and essentially a silent failure if we don't have a token?

---

@RoryO yeah, nil in this case means the uid can not be resolved from the token.  Not a failure, because the absence of it means the user is not authenticated (and allows us to fall back on the session strategy for the moment).